src/scanner.c

```diff
@@ -945,8 +969,17 @@ static UInt NextSymbol(void)
 
     // skip over <spaces>, <tabs>, <newlines> and comments
     while (c == ' ' || c == '\t' || c== '\n' || c== '\r' || c == '\f' || c=='#') {
-        if (c == '#')
+        if (c == '#'){
+            c = GET_NEXT_CHAR_NO_LC();
```

Comment:
I am not sure I understand how this is correct... what if the lines ends right after the `#`? Then this command moves us to the start of the next line, and then `SKIP_TO_END_OF_LINE();` skips *that* line, doesn't it? What am I missing?

Reply:
The problem I am trying to tackling is that there is very specifically [this single comment](https://github.com/gap-system/gap/blob/master/lib/stbcbckt.gi#L207).

But if your current character is a linebreak, `SKIP_TO_END_OF_LINE` does not do anything.